bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/handler/RotelHandler.java

```diff
@@ -0,0 +1,2006 @@
+ * The {@link RotelHandler} is responsible for handling commands, which are sent to one of the channels.
+ *
+ * @author Laurent Garnier - Initial contribution
+ */
+@NonNullByDefault
+public class RotelHandler extends BaseThingHandler implements RotelMessageEventListener {
+
+    private final Logger logger = LoggerFactory.getLogger(RotelHandler.class);
+
+    private static final RotelModel DEFAULT_MODEL = RotelModel.RSP1066;
+    private static final long POLLING_INTERVAL = TimeUnit.SECONDS.toSeconds(60);
+
+    private @Nullable ScheduledFuture<?> reconnectJob;
+    private @Nullable ScheduledFuture<?> powerOnJob;
+    private @Nullable ScheduledFuture<?> powerOffJob;
+    private @Nullable ScheduledFuture<?> powerOnZone2Job;
+    private @Nullable ScheduledFuture<?> powerOnZone3Job;
+    private @Nullable ScheduledFuture<?> powerOnZone4Job;
+
+    private RotelStateDescriptionOptionProvider stateDescriptionProvider;
+    private SerialPortManager serialPortManager;
+
+    private RotelConnector connector = new RotelSimuConnector(DEFAULT_MODEL, RotelProtocol.HEX);
+
+    private boolean simu;
+
+    private int minVolume;
+    private int maxVolume;
+    private int maxToneLevel;
+
+    private int currentZone = 1;
+    private boolean selectingRecord;
+    private @Nullable Boolean power;
+    private boolean powerZone2;
+    private boolean powerZone3;
+    private boolean powerZone4;
+    private RotelSource source = RotelSource.CAT0_CD;
+    private @Nullable RotelSource recordSource;
+    private @Nullable RotelSource sourceZone2;
+    private @Nullable RotelSource sourceZone3;
+    private @Nullable RotelSource sourceZone4;
+    private RotelDsp dsp = RotelDsp.CAT1_NONE;
+    private int volume;
+    private boolean mute;
+    private boolean fixedVolumeZone2;
+    private int volumeZone2;
+    private boolean muteZone2;
+    private boolean fixedVolumeZone3;
+    private int volumeZone3;
+    private boolean muteZone3;
+    private boolean fixedVolumeZone4;
+    private int volumeZone4;
+    private boolean muteZone4;
+    private int bass;
+    private int treble;
+    private RotelPlayStatus playStatus = RotelPlayStatus.STOPPED;
+    private int track;
+    private double frequency;
+    private String frontPanelLine1 = "";
+    private String frontPanelLine2 = "";
+    private int brightness;
+
+    private Object sequenceLock = new Object();
+
+    /**
+     * Constructor
+     */
+    public RotelHandler(Thing thing, RotelStateDescriptionOptionProvider stateDescriptionProvider,
+            SerialPortManager serialPortManager) {
+        super(thing);
+        this.stateDescriptionProvider = stateDescriptionProvider;
+        this.serialPortManager = serialPortManager;
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Start initializing handler for thing {}", getThing().getUID());
+
+        RotelModel rotelModel = DEFAULT_MODEL;
+        if (getThing().getThingTypeUID().equals(THING_TYPE_RSP1066)) {
```

Comment:
I can't directly because getThing().getThingTypeUID() is of type ThingTypeUID.
But ok I can define constants for each thing type ID which is a string.

Reply:
Done